File: app/src/main/java/com/blues/module_gank/old/model_article/ArticleFragment.java
package com.blues.module_gank.old.model_article;

import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import com.blues.R;
import com.blues.WebViewDialog;
import com.blues.framework.base.BaseFragment;
import com.blues.framework.utils.FileUtil;
import com.google.gson.Gson;
import com.jude.easyrecyclerview.EasyRecyclerView;
import com.jude.easyrecyclerview.adapter.BaseViewHolder;
import com.jude.easyrecyclerview.adapter.RecyclerArrayAdapter;

import androidx.recyclerview.widget.LinearLayoutManager;

public class ArticleFragment extends BaseFragment {

    private RecyclerArrayAdapter<Article.ArticleBean> mAdapter;

    @Override
    public int setLayoutResourceId() {
        return R.layout.fragment_article;
    }

    @Override
    public void initLayout(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        EasyRecyclerView revVideo = rootView.findViewById(R.id.ev_article);
        revVideo.setLayoutManager(new LinearLayoutManager(getmContext()));
        revVideo.setAdapter(mAdapter = new RecyclerArrayAdapter<Article.ArticleBean>(getmContext()) {
            @Override
            public BaseViewHolder OnCreateViewHolder(ViewGroup parent, int viewType) {
                return new ArticleHolder(parent);
            }
        });
        String io = FileUtil.getAssetsFile(getmContext(), "article.json");
        Article article = new Gson().fromJson(io, Article.class);
        Log.i("Blues", article.toString());
        mAdapter.addAll(article.getArticle());
        mAdapter.notifyDataSetChanged();

        mAdapter.setOnItemClickListener(position -> new WebViewDialog(getmContext(), article.getArticle().get(position).getUrl()).show());
    }

    @Override
    public void lazyFetchData() {

    }

    @Override
    public void setListener() {

    }
}

File: app/src/main/java/com/blues/gankio/v1/view/ArticleFragment.kt
package com.blues.gankio.v1.view

import com.blues.framework.utils.FileUtil.getAssetsFile
import com.jude.easyrecyclerview.adapter.RecyclerArrayAdapter
import com.blues.R
import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import androidx.lifecycle.lifecycleScope
import com.jude.easyrecyclerview.EasyRecyclerView
import androidx.recyclerview.widget.LinearLayoutManager
import com.google.gson.Gson
import com.blues.framework.dialog.WebViewDialog
import com.blues.framework.base.BaseKoinFragment
import com.blues.framework.base.catch
import com.blues.framework.utils.HelperUtil
import com.blues.gankio.v1.model.ArticleBean
import com.jude.easyrecyclerview.adapter.BaseViewHolder
import kotlinx.coroutines.flow.channelFlow
import kotlinx.coroutines.flow.collect
import kotlinx.coroutines.launch

class ArticleFragment : BaseKoinFragment() {

    private lateinit var mAdapter: RecyclerArrayAdapter<ArticleBean.ArticleDetailBean>

    override fun getLayoutId(): Int = R.layout.fragment_article

    override fun initData(inflater: LayoutInflater, container: ViewGroup?, saveInstanced: Bundle?) {

        rootView.findViewById<EasyRecyclerView>(R.id.ev_article).apply {
            setLayoutManager(LinearLayoutManager(requireContext()))
            adapter = object :
                RecyclerArrayAdapter<ArticleBean.ArticleDetailBean>(requireContext()) {
                override fun OnCreateViewHolder(
                    parent: ViewGroup,
                    viewType: Int
                ): BaseViewHolder<*> {
                    return ArticleHolder(parent)
                }
            }.also { mAdapter = it }
        }

        lifecycleScope.launchWhenStarted {
            channelFlow<String> {
                getAssetsFile(requireContext(), "article.json")
            }.catch {
                HelperUtil.showSimpleLog(localizedMessage)
            }.collect {
                val articleBean = Gson().fromJson(it, ArticleBean::class.java)
                mAdapter.apply {
                    addAll(articleBean.articleDetail)
                    notifyDataSetChanged()
                    setOnItemClickListener { position: Int ->
                        WebViewDialog(
                            requireContext(),
                            articleBean.articleDetail[position].url
                        ).show()
                    }
                }
            }
        }
//        val io = getAssetsFile(requireContext(), "article.json")
//        val article = Gson().fromJson(io, ArticleBean::class.java)
//        mAdapter.apply {
//            addAll(article.articleDetail)
//            notifyDataSetChanged()
//            setOnItemClickListener { position: Int -> WebViewDialog(requireContext(), article.articleDetail[position].url).show() }
//        }
    }
}